state_test.go

```diff
@@ -183,8 +207,13 @@ func TestMemberList_ProbeNode_Suspect_Dogpile(t *testing.T) {
 				peer := HostMemberlist(peerAddr.String(), t, func(c *Config) {
 					c.BindPort = bindPort
 				})
-				defer peer.Shutdown()
+				defer func() {
+					if err := peer.Shutdown(); err != nil {
+						t.Fatal(err)
+					}
+				}()
 
+				//nolint:staticcheck // reason: peers used later indirectly
 				peers = append(peers, peer)
```

Comment:
`peers` is being appended to in this loop. But I don't see the array itself being used elsewhere. Where is it being used indirectly?